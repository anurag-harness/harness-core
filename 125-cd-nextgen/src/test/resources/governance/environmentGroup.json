{
  "envGroupRef": "env_group_1",
  "metadata": {
    "parallel": true
  },
  "deployToAll": false,
  "environments": [
    {
      "environmentRef": "my_environment",
      "infrastructureDefinitions": [
        {
          "identifier": "my_infra_with_ns_runtime",
          "inputs": {
            "identifier": "my_infra_with_ns_runtime",
            "type": "KubernetesDirect",
            "spec": {
              "namespace": "harness",
              "releaseName": "dev-0001"
            }
          }
        }, {
          "identifier": "some-<+expression>-with_should_be_ignored"
        },
        {
          "identifier": "my_infra"
        }
      ]
    },
    {
      "environmentRef": "non_existent_environment",
      "infrastructureDefinitions": [
        {
          "identifier": "some_infra"
        }
      ]
    }
  ]
}