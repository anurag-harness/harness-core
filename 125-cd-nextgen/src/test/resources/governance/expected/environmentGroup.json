{
  "identifier" : "env_group_1",
  "metadata" : {
    "parallel" : true
  },
  "environments" : [ {
    "environmentRef" : "my_environment",
    "identifier" : "my_environment",
    "name" : "dev_environment",
    "description" : "description",
    "type" : "PreProduction",
    "accountIdentifier" : "accountId",
    "orgIdentifier" : "orgId",
    "projectIdentifier" : "projectId",
    "tags" : {
      "prod" : "false"
    },
    "infrastructures" : [ {
      "type" : "KubernetesDirect",
      "spec" : {
        "namespace" : "harness",
        "releaseName" : "dev-0001",
        "connector" : {
          "name" : "k8s_connector_name",
          "identifier" : "k8s_connector",
          "description" : null,
          "orgIdentifier" : null,
          "projectIdentifier" : null,
          "tags" : null,
          "type" : "K8sCluster",
          "spec" : {
            "credential" : null,
            "delegateSelectors" : [ "del-1" ]
          }
        }
      }
    }, {
      "type" : "ECS",
      "spec" : {
        "cluster" : "us-east-1-cluster",
        "region" : "us-east-1"
      }
    } ]
  }, { } ],
  "name" : "my environment group",
  "deployToAll" : false
}